From 862385ce2eb6eeaf893e44a2227b0bf9626cb91a Mon Sep 17 00:00:00 2001
From: Andy CrossGate Yan <GeForce8800Ultra@gmail.com>
Date: Tue, 11 Aug 2026 09:53:05 +0000
Subject: [PATCH] SystemUI: Follow monet theme on battery chip

Why does this chip even exist...

Change-Id: I178fdda5d8d310c4c7fa5f05f9911b3f94986cb9

[Port to A16] A16 BatteryStatusChip dropped settingslib flags import block and BatteryColors import; Utils import added before BatteryMeterView import. updateResources body identical.

Change-Id: I42913322947a491901f2866ac88a979ba42c4f45
---
 packages/SystemUI/res/drawable/statusbar_chip_bg.xml      | 8 +++-----
 .../com/android/systemui/statusbar/BatteryStatusChip.kt   | 6 ++++++
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/packages/SystemUI/res/drawable/statusbar_chip_bg.xml b/packages/SystemUI/res/drawable/statusbar_chip_bg.xml
index d7de16d7c5bb..dd1db04e77d7 100644
--- a/packages/SystemUI/res/drawable/statusbar_chip_bg.xml
+++ b/packages/SystemUI/res/drawable/statusbar_chip_bg.xml
@@ -15,9 +15,7 @@
      limitations under the License.
 -->
 
-<shape
-    xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:androidprv="http://schemas.android.com/apk/prv/res/android">
-    <solid android:color="?androidprv:attr/colorAccentPrimary" />
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="@color/privacy_chip_background" />
     <corners android:radius="@dimen/ongoing_appops_chip_bg_corner_radius" />
-</shape>
\ No newline at end of file
+</shape>
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/BatteryStatusChip.kt b/packages/SystemUI/src/com/android/systemui/statusbar/BatteryStatusChip.kt
index bff15bd8922f..6a60947d2a9d 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/BatteryStatusChip.kt
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/BatteryStatusChip.kt
@@ -22,6 +22,7 @@ import android.util.AttributeSet
 import android.view.View
 import android.widget.FrameLayout
 import android.widget.LinearLayout
+import com.android.settingslib.Utils
 import com.android.systemui.battery.BatteryMeterView
 import com.android.systemui.res.R
 import com.android.systemui.statusbar.core.NewStatusBarIcons
@@ -68,6 +69,11 @@ class BatteryStatusChip @JvmOverloads constructor(context: Context, attrs: Attri
 
     @SuppressLint("UseCompatLoadingForDrawables")
     private fun updateResources() {
+        val primaryColor =
+            Utils.getColorAttrDefaultColor(context, android.R.attr.textColorPrimaryInverse)
+        val textColorSecondary =
+            Utils.getColorAttrDefaultColor(mContext, android.R.attr.textColorSecondaryInverse)
+        batteryMeterView.updateColors(primaryColor, textColorSecondary, primaryColor)
         roundedContainer.background = mContext.getDrawable(R.drawable.statusbar_chip_bg)
     }
 }
-- 
2.43.0

